Content.Server/_Shitcode/Heretic/EntitySystems/PathSpecific/AristocratSystem.cs

```diff
@@ -74,55 +207,134 @@ public override void Update(float frameTime)
+            return;
 
-        FreezeAtmos(ent);
+        var coords = Transform(ent).Coordinates;
+        var step = ent.Comp.UpdateStep;
 
-        DoChristmas(ent, lookup);
+        switch (step)
+        {
+            case 0:
+                ExtinguishFires(ent, coords);
+                break;
+            case 1:
+                FreezeAtmos(ent);
+                break;
+            case 2:
+                DoChristmas(ent, coords);
+                break;
+            case 3:
+                SpookyLights(ent, coords);
+                break;
+            case 4:
+                FreezeNoobs(ent, coords);
+                break;
+            default:
+                ent.Comp.UpdateStep = 0;
+                break;
+        }
 
-        FreezeNoobs(ent, lookup);
+        ent.Comp.UpdateStep++;
     }
 
     // makes shit cold
     private void FreezeAtmos(Entity<AristocratComponent> ent)
     {
         var mix = _atmos.GetTileMixture((ent, Transform(ent)));
+        var freezingTemp = Atmospherics.T0C;
+
         if (mix != null)
-            mix.Temperature -= 50f;
+        {
+            if (mix.Temperature > freezingTemp)
+                mix.Temperature = freezingTemp;
+
+            mix.Temperature -= 100f;
+        }
     }
 
-    // replaces certain things with their winter analogue
-    private void DoChristmas(Entity<AristocratComponent> ent, HashSet<EntityUid> lookup)
+    // extinguish gases on tiles
+    private void ExtinguishFiresTiles(Entity<AristocratComponent> ent)
+    {
+        var tilerefs = GetTiles(ent);
+
+        if (tilerefs == null)
+            return;
+
+        if (tilerefs.Count == 0)
+            return;
+
+        foreach (var tile in tilerefs)
+        {
+            _atmos.HotspotExtinguish(tile.GridUid, tile.GridIndices);
+        }
+    }
+
+    // extinguish ppl and stuff
+    private void ExtinguishFires(Entity<AristocratComponent> ent, EntityCoordinates coords)
+    {
+        var fires = _lookup.GetEntitiesInRange<FlammableComponent>(coords, ent.Comp.Range);
+
+        foreach (var entity in fires)
+        {
+            if (entity.Comp.OnFire)
+                _flammable.Extinguish(entity);
+        }
+
+        ExtinguishFiresTiles(ent);
+    }
+
+    // replaces certain things with their winter analogue (amongst other things)
+    private void DoChristmas(Entity<AristocratComponent> ent, EntityCoordinates coords)
     {
         SpawnTiles(ent);
 
-        foreach (var look in lookup)
+        var dspec = new DamageSpecifier();
+        dspec.DamageDict.Add("Structural", 100);
+
+        var tags = _lookup.GetEntitiesInRange<TagComponent>(coords, ent.Comp.Range);
+
+        foreach (var tag in tags)
         {
-            if (TryComp<TagComponent>(look, out var tag))
+            // walls
+            if (_tag.HasTag(tag.Owner, "Wall") && _rand.Prob(.45f)
+                && Prototype(tag) != null && Prototype(tag)!.ID != SnowWallPrototype)
```

Comment:
The prob should probably be a cvar but i aint getting picky.
```suggestion
            if (_tag.HasTag(tag.Owner, "Wall")
                && _rand.Prob(.45f)
                && Prototype(tag) != null
                && Prototype(tag)!.ID != SnowWallPrototype)

Reply:
so true sigma